src/CSharp/CodeCracker/Refactoring/XmlDocumentationAnalyzer.cs

```diff
@@ -0,0 +1,65 @@
+        internal const string Category = SupportedCategories.Refactoring;
+
+        internal static DiagnosticDescriptor XmlDocumentationRule = new DiagnosticDescriptor(
+            DiagnosticId.XmlDocumentation.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            SupportedCategories.Refactoring,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            helpLinkUri: HelpLink.ForDiagnostic(DiagnosticId.XmlDocumentation));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(XmlDocumentationRule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(Analyzer, SyntaxKind.SingleLineDocumentationCommentTrivia);
+
+        private static void Analyzer(SyntaxNodeAnalysisContext context)
+        {
+            var documentationNode = (DocumentationCommentTriviaSyntax)context.Node;
+            var method = documentationNode.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().First();
+            var methodParameters = method.ParameterList.Parameters;
+            var xElementsWitAttrs = documentationNode.Content.OfType<XmlElementSyntax>()
+                                    .Where(xEle => xEle.StartTag.Name.LocalName.ValueText == "param")
+                                    .SelectMany(xEle => xEle.StartTag.Attributes, (xEle, attr) => (XmlNameAttributeSyntax)attr);
+
+            var methodParameterLookUp = methodParameters.ToLookup(parameter => parameter.Identifier.ValueText);
+            var docParameterLookUp = xElementsWitAttrs.ToLookup(attr => attr.Identifier.Identifier.ValueText);
+
+            var keys = methodParameterLookUp.Select(p => p.Key).Union(docParameterLookUp.Select(p => p.Key)).ToImmutableHashSet();
+
+            var paramterWithDocParameter = (from key in keys
+                                            let Parameter = methodParameterLookUp[key].FirstOrDefault()
+                                            let DocParameter = docParameterLookUp[key].FirstOrDefault()
+                                            select new { Parameter, DocParameter });
+
+            if (paramterWithDocParameter.Any(p => p.Parameter == null))
+            {
+                var properties = new Dictionary<string, string> {["kind"] = "unexistent" }.ToImmutableDictionary();
+                var diagnostic = Diagnostic.Create(XmlDocumentationRule, documentationNode.GetLocation(), properties, properties["kind"]);
+                context.ReportDiagnostic(diagnostic);
+            }
+
+            if (paramterWithDocParameter.Any(p => p.DocParameter == null))
+            {
+                var properties = new Dictionary<string, string> { ["kind"] = "missing" }.ToImmutableDictionary();
+                var diagnostic = Diagnostic.Create(XmlDocumentationRule, documentationNode.GetLocation(), properties, properties["kind"]);
```

Comment:
I am finding it difficult to follow the `missing` and `unexistent` texts. They almost mean the same thing. Can you try to be more descriptive? Maybe `missingDoc` and `nonexistentParam`, or something like that?